src/main/java/edu/harvard/iq/dataverse/DatasetPage.java

```diff
@@ -6144,7 +6144,7 @@ public String getWebloaderUrlForDataset(Dataset d) {
     String signpostingLinkHeader = null;
 
     public String getSignpostingLinkHeader() {
-        if (!workingVersion.isReleased()) {
+        if ((workingVersion==null) || (!workingVersion.isReleased())) {
```

Comment:
Is this a complete fix for #9954?

@haarli is reporting that with this null check in place another NPE emerges: https://github.com/IQSS/dataverse/issues/9954#issuecomment-1734924704

Reply:
It should be now (with the change in the refresh() method as well.) Note that the fixes don't appear to be needed in 6.x, but the null checks that are in this PR will avoid having the page fail if the problem recurs (and will throw a warning in the log).

Reply:
Right. Fixed in c9c6cf2. Thanks.